### DISCUSSION TITLE
How do I render a table of components where the size of the table is determined at runtime?

### DISCUSSION THREAD

**Discussion by InnocentZero** [ArthurSonzogni/FTXUI#866] [Q&A]
Pretty much the title. I am in need of an interactive table. At the moment, I'm using the grid-container component that is available outside the repository. However, it is wasting a lot of space in between the gaps. So I wanted to implement something similar to [this](https://github.com/ArthurSonzogni/FTXUI/discussions/568#discussioncomment-4888054), but cannot figure out a way to do so for a table in runtime. Here are the relevant parts of the implementation:

```cpp
ftxui::Component Disp::Cell(std::string &label, const ftxui::Color &fg,
                            const ftxui::Color &hi) {
    using namespace ftxui;
    auto cell = Renderer([&](bool focused) {
        auto style = [](Element Elem) {
            return Elem | center | flex | size(WIDTH, EQUAL, 5) |
                   size(HEIGHT, EQUAL, 5);
        };
        if (focused) {
            return text(label) | bold | bgcolor(hi) | style |
                   borderStyled(BorderStyle::EMPTY);
            ;
        } else {
            return text(label) | bgcolor(Color::DarkRed) | style |
                   borderStyled(BorderStyle::EMPTY);
        }
    });
    return cell;
}
```

and somewhere else

```cpp
    std::vector<Components> buttons(16);
    for (std::uint8_t i = 0; i < 16; i++) {
        for (std::uint8_t j = 0; j < 16; j++) {
            buttons[i].push_back(Cell(table.labels[16 * i + j],
                                      table.fgs[16 * i + j],
                                      table.his[16 * i + j]) |
                                 CatchEvents({i, j}) | center | flex);
        }
    }
```

Any help?

**Comment by InnocentZero** [comment]
The current value of 16 is a placeholder. I plan to add runtime values later on.

**Reply by pat-RR-ick** [reply]
this might help https://github.com/pat-RR-ick/ftxui-trading-terminal-example

**Reply by pat-RR-ick** [reply]
check the orderbook component

**Reply by SLHome** [reply]
This example shows dynamic content at a fixed size.
I join the question, I’m interested in how to implement dynamic size.
